### PR TITLE
Support redirects between GH namespaces

### LINKED ISSUE #17: Redirect hashicorp providers to the forked version artifacts
We should be hosting opentf forks of each of the hashicorp providers and in the registry, anytime someone requests a `hashicorp` package, we should redirect it to the opentf versions.

### PR DESCRIPTION
**Pull request by marcinwyszynski**
Closes #17

If a particular namespace does not publish providers in the GitHub releases, an environment variable `PROVIDER_NAMESPACE_REDIRECTS` allows mapping it to a namespace that mirrors it.

### CODE REVIEW

File: lambda/router.go [hunk 1]
```diff
@@ -13,6 +13,15 @@ import (
 type Config struct {
 	ManagedGithubClient *github.Client
 	RawGithubv4Client   *githubv4.Client
+	Redirects           map[string]string
+}
+
+func (c Config) EffectiveNamespace(namespace string) string {
```
Comment (Yantrio) [inline lambda/router.go]: Let's add a comment here to denote what this is doing and why its needed

```suggestion
// EffectiveNamespace will map namespaces for providers and modules in situations where the author (owner of the namespace) does not release artifacts as github releases.
func (c Config) EffectiveNamespace(namespace string) string {
```

**Review by Yantrio — APPROVED** [review]
one small comment nit, apart from that it LGTM 👍

File: lambda/router.go (hunk 1 above)
Comment (marcinwyszynski) [reply lambda/router.go]: @Yantrio we don't use it for modules.

**Comment by marcinwyszynski** [comment]
@Yantrio thanks, I added comments and removed the ambiguity to point out that these rewrites are only for providers.

**Pull request merged**